src/ir/ir_traversing_visitor.h

```diff
@@ -16,55 +16,93 @@
 #ifndef SRC_IR_IR_TRAVERSING_VISITOR_H_
 #define SRC_IR_IR_TRAVERSING_VISITOR_H_
 
+#include "src/common/utils/fold.h"
+#include "src/common/utils/types.h"
 #include "src/ir/ir_visitor.h"
 #include "src/ir/module.h"
 
 namespace raksha::ir {
 
+template<class Result, class Enable = void>
+struct DefaultValueGetter {
+  static Result Get() { LOG(FATAL) << "Override required for non-default-constructible type."; }
+};
```

Comment:
Leave this for now, but I wonder if we can make this a compile time error rather than runtime error. Can you file an issue?

 I found some tricks to make SFINAE errors more readable. See https://gracicot.github.io/tricks/2017/07/01/deleted-function-diagnostic.html. Can you also mention this link in the filed issue?

Reply:
Opened https://github.com/google-research/raksha/issues/618 and added comment with link.